google/cloud/documentai_toolbox/wrappers/page.py

```diff
@@ -90,97 +106,184 @@ def sample_table_to_csv():
+class Block:
+    """Represents a wrapped documentai.Document.Page.Block.
 
     Attributes:
-        documentai_line (google.cloud.documentai.Document.Page.Line):
-            Required. The original google.cloud.documentai.Document.Page.Line object.
+        documentai_object (google.cloud.documentai.Document.Page.Block):
+            Required. The original google.cloud.documentai.Document.Page.Block object.
         text (str):
             Required. UTF-8 encoded text.
     """
 
-    documentai_line: documentai.Document.Page.Line
-    text: str
+    documentai_object: documentai.Document.Page.Block = dataclasses.field(
+        init=True, repr=False
+    )
+    text: str = dataclasses.field(init=True, repr=False)
+    _page: Page = dataclasses.field(init=True, repr=False)
 
+    paragraphs: List[Paragraph] = dataclasses.field(init=False, repr=False)
 
-@dataclasses.dataclass
-class FormField:
-    """Represents a wrapped documentai.Document.Page.FormField.
+    def __post_init__(self):
+        self.paragraphs = _get_children_of_element(
+            self.documentai_object, self._page.paragraphs
+        )
+
+
+def _table_wrapper_from_documentai_table(
+    documentai_table: documentai.Document.Page.Table, text: str
+) -> Table:
+    r"""Returns a Table.
+
+    Args:
+        documentai_table (documentai.Document.Page.Table):
+            Required. A documentai.Document.Page.Table.
+        text (str):
+            Required. UTF-8 encoded text in reading order
+            from the document.
+
+    Returns:
+        Table:
+            A Table.
 
-    Attributes:
-        documentai_formfield (google.cloud.documentai.Document.Page.FormField):
-            Required. The original google.cloud.documentai.Document.Page.FormField object.
-        field_name (str):
-            Required. The form field name
-        field_value (str):
-            Required. The form field value
     """
 
-    documentai_formfield: documentai.Document.Page.FormField
-    field_name: str
-    field_value: str
+    header_rows = _table_rows_from_documentai_table_rows(
+        table_rows=list(documentai_table.header_rows), text=text
+    )
+    body_rows = _table_rows_from_documentai_table_rows(
+        table_rows=list(documentai_table.body_rows), text=text
+    )
+
+    return Table(
+        documentai_table=documentai_table, body_rows=body_rows, header_rows=header_rows
+    )
+
+
+def _get_hocr_bounding_box(
+    element_with_layout: ElementWithLayout,
+    dimension: documentai.Document.Page.Dimension,
+) -> str:
+    r"""Returns a hOCR bounding box string.
+
+    Args:
+        element_with_layout (ElementWithLayout):
+            Required. an element with layout fields.
+        dimension (documentai.Document.Page.Dimension):
+            Required. Page dimension.
+
+    Returns:
+        str:
+            hOCR bounding box sring.
+    """
+
+    if element_with_layout.layout.bounding_poly.vertices:
+        min_x, min_y = (
```

Comment:
Let's refactor this duplicate code (with the code below) into something that can be easily unit-tested just once.

Reply:
How can we refactor this more? We're dealing with two separate objects `vertices` and `normalized vertices`

Reply:
Maybe something like the following?

```
def _get_xy(..., normalized: bool=False, min: bool=False):
    ...
```

To be sure, such a function will still need to be tested for all different combinations of use cases (`normalized=True`, `normalized=False` etc) perhaps in a parametrized test.